src/core/Akka/Actor/SupervisorStrategy.cs

```diff
@@ -245,8 +245,7 @@ public OneForOneStrategy(Func<Exception, Directive> decider) : this(-1, -1, deci
         /// <summary>
         /// Serialization-friendly constructor
         /// </summary>
-        [Obsolete("For serialization purposes only!")]
-        public OneForOneStrategy() : this(DefaultDecider) { }
+        protected OneForOneStrategy() : this(DefaultDecider) { }
```

Comment:
Removed public obsolete hack in favor for protected ctor now that the json serializer supports this.